runtime/src/module.c

```diff
@@ -555,8 +556,18 @@ module_new_from_json(char *file_name)
 			                                   relative_deadline_us, port, request_size, response_size,
 			                                   admissions_percentile, expected_execution_us);
 			if (module == NULL) goto module_new_err;
-
+			
 			assert(module);
+
+			if (tail_module == NULL) {
```

Comment:
We'll need to come up with a json representation for this, so I know this isn't the "final" form of this code. However, I wanted to point out that these conditionals are perhaps a little more complex than they have to be. I believe you can replace the conditionals with:

```c
module->next_module      = NULL;
tail_module->next_module = module;
tail_module              = module;
```

Reply:
but tail_module is initialized to NULL, no allocated memory, tail_module->next_module is not accessible.

Reply:
Thanks, I figured I missed a case. Maybe focus on only that difference, keeping most code shared, then?

```c
if (tail_module != NULL) tail_module->next_module = module;
tail_module->next_module = NULL;
tail_module              = module;
```
Point being: I had to read this code about 5 times before I understood it. And I still got the reformat wrong ;-) It feels like it can be simplified.

Reply:
> Thanks, I figured I missed a case. Maybe focus on only that difference, keeping most code shared, then?
> 
> ```c
> if (tail_module != NULL) tail_module->next_module = module;
> tail_module              = module;
> tail_module->next_module = NULL;
> ```
> 
> Point being: I had to read this code about 5 times before I understood it. And I still got the reformat wrong ;-) It feels like it can be simplified.

I totally agree, this is more clear/readable than my version, I will change the code like this, sorry for the 5 times reading:)